packages/core/src/domain/tracekit/tracekit.ts

```diff
@@ -55,36 +57,44 @@ export function startUnhandledErrorCollection(callback: UnhandledErrorCallback)
  */
 function instrumentOnError(callback: UnhandledErrorCallback) {
   return instrumentMethodAndCallOriginal(window, 'onerror', {
-    before(this: any, message: Event | string, url?: string, lineNo?: number, columnNo?: number, errorObj?: Error) {
-      let stack: StackTrace
-
-      if (errorObj) {
-        stack = computeStackTrace(errorObj)
+    before(this: any, message: Event | string, url?: string, lineNo?: number, columnNo?: number, errorObj?: unknown) {
+      // Priority is given to errorObj if it is an instance of Error. This allows us to compute the most comprehensive stack trace.
+      if (errorObj instanceof Error) {
+        const stack = computeStackTrace(errorObj)
         callback(stack, errorObj)
       } else {
-        const location = {
-          url,
-          column: columnNo,
-          line: lineNo,
-        }
+        let name: string | undefined
+        let msg: string | undefined
 
-        let name
-        let msg = message
+        // Try to extract information about the error from the provided message
         if ({}.toString.call(message) === '[object String]') {
-          const groups = ERROR_TYPES_RE.exec(msg as string)
+          msg = message as string
+          const groups = ERROR_TYPES_RE.exec(message as string)
           if (groups) {
             name = groups[1]
             msg = groups[2]
           }
         }
 
-        stack = {
-          name,
-          message: typeof msg === 'string' ? msg : undefined,
-          stack: [location],
+        // If an errorObj is provided, give priority to its content to populate the message of the stack trace
+        if (errorObj) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** use an `else if` to be explicit on whether the `if ({}.toString.call(message) === '[object String]') {` condition or the `if(errorObj)` condition takes precedence to define the `msg` variable